File: ruby/contains_nearby_duplicate.rb
# @param {Integer[]} nums
# @param {Integer} k
# @return {Boolean}
def contains_nearby_duplicate(nums, k)
    seen = {}
    
    nums.each_with_index do |num, i|
        if seen[num] && (i - seen[num]) <= k
            return true
        end
        
        seen[num] = i
    end
    
    false
end

File: typescript/containsNearbyDuplicate.ts
function containsNearbyDuplicate(nums: number[], k: number): boolean {
  const seen: { [key: number]: number } = {};

  for (let i = 0; i < nums.length; i++) {
    const num = nums[i];

    if (seen[num] !== undefined && (i - seen[num]) <= k) {
      return true;
    }

    seen[num] = i;
  }

  return false;
}
